### PR TITLE
Adding support for spark3.0

### PR DESCRIPTION
**Pull request by uday1409**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -358,7 +359,20 @@
...
-
+        <profile>
+            <id>spark-3.0.0</id>
+            <properties>
+                <spark.version>3.0.0</spark.version>
+                <json4s.version>3.6.6</json4s.version>
+            </properties>
+        </profile>
+        <profile>
+            <id>spark-2.4.2</id>
```
Comment (Zejnilovic) [inline pom.xml:370]: Why `2.4.2`? It was `2.4.4` and could maybe be bumped up. 

Same for `json4s`

Comment (uday1409) [reply pom.xml:370]: lemme check and resolve. Ideally anything 2.4.x should not be an issue , but will revert

File: pom.xml [hunk 2]
```diff
@@ -358,7 +359,20 @@
                 <scala.version>${scala_2.13.version}</scala.version>
             </properties>
         </profile>
-
+        <profile>
+            <id>spark-3.0</id>
+            <properties>
+                <spark.version>3.0.0</spark.version>
```
Comment (Zejnilovic) [inline pom.xml:365]: There is also spark v 3.0.1. 
 Release notes - https://spark.apache.org/releases/spark-release-3-0-1.html

**Comment by wajda** [comment]
Assuming Spark 3.0 support implies Spline, as I explained in the #27 it has to be solved in either Spline agent alone, or split between Spline agent and commons in a way that keeps Commons decoupled from Spline or Spark.

**Pull request closed without merging**